src/operations.cpp

```diff
@@ -1399,6 +1419,67 @@ void last_write_time(const path& p, const std::time_t new_time,
 # endif
 }
 
+BOOST_FILESYSTEM_DECL
+std::time_t creation_time(const path& p, system::error_code* ec)
+{
+# ifdef BOOST_POSIX_API
+# ifdef __linux__
+  struct statx path_stat;
+  if (error(::statx(AT_FDCWD, p.c_str(), AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT |
+	AT_STATX_SYNC_AS_STAT, STATX_BTIME, &path_stat)!= 0 ? BOOST_ERRNO : 0,
+    p, ec, "boost::filesystem::creation_time"))
+      return std::time_t(-1);
+  return std::time_t(path_stat.stx_btime.tv_sec);
+# elif defined(__APPLE__)
+  // The stat64() system calls first appeared in Mac OS X 10.5 (Leopard)
+#   if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
```

Comment:
I may be missing something, but it looks like when this condition is not satisfied, the function is left empty.

If `st_birthtimespec` is not available on older versions of Mac OS X then the function must return `ENOSYS`.